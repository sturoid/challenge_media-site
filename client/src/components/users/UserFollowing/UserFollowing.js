import React from 'react';
import PropTypes from 'prop-types';
import Table from '../../_shared/Table';

const UserFollowing = ({ following }) => (
  <Table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Location</th>
        <th>Occupation</th>
      </tr>
    </thead>
    <tbody>
      {following.map(f => (
        <tr key={f.id}>
          <td>{f.name}</td>
          <td>{f.location}</td>
          <td>{f.occupation}</td>
        </tr>
      ))}
    </tbody>
  </Table>
);

UserFollowing.propTypes = {
  following: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      occupation: PropTypes.string.isRequired
    })
  ).isRequired
};

export default UserFollowing;
