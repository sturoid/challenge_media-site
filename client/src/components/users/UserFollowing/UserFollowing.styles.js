import { css } from 'emotion';

const follower = css`
  display: flex;
  align-items: center;
  & > div {
    width: 200px;
    padding-left: 10px;
  }
  & > div:not(:first-child) {
    padding-top: 10px;
  }
`;

export default follower;
