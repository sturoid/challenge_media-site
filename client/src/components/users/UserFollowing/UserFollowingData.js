import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import Loader from '../../../components/_shared/Loader';
import UserFollowing from './UserFollowing';

class UserFollowersData extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      error: false,
      following: []
    };
  }

  componentDidMount() {
    this.fetchDataAndSetState();
  }

  async fetchDataAndSetState() {
    const { data: following } = await axios.get(
      `/api/v1/user/${this.props.userId}/following`
    );
    this.setState({ following });
  }

  render() {
    const { loading, error, following } = this.state;
    if (loading) return <Loader loading={loading} />;
    if (error) return <div>{error.message}</div>;
    return <UserFollowing following={following} />;
  }
}

UserFollowersData.propTypes = {
  userId: PropTypes.string.isRequired
};

export default UserFollowersData;
