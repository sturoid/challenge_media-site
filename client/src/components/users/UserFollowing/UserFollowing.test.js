import React from 'react';
import renderer from 'react-test-renderer';
import UserFollowing from './UserFollowing';
// import followersData from '../../../../../lib/test-data/userFollowers';

describe('<UserFollowing />', () => {
  it('renders correctly and equals snapshot', () => {
    const tree = renderer.create(<UserFollowing />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
