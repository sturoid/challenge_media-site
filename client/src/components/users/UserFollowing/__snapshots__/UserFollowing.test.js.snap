// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserFollowing /> renders correctly and equals snapshot 1`] = `
<div
  className="css-qijqp5"
>
  <table
    className="css-1z0wrym"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Location
        </th>
        <th>
          Occupation
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Gage
        </td>
        <td>
          
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Kristin
        </td>
        <td>
          
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Florian
        </td>
        <td>
          Interaction Design Student
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Amy Moore
        </td>
        <td>
          First Assistant
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
